### DISCUSSION TITLE
Calendar component USP

### DISCUSSION THREAD

**Discussion by mmailaender** [formaat-design/reshaped#120] [General]
Hi Dmitry! 😊

Currently checking your notion board – a lot of interesting new stuff there! I also saw the [calendar issue](https://blvworkspace.notion.site/64cf1f5713344a7383330e0402f43949?v=b88a6dbbcb9a4faeb867d40d09ec0b12&p=2429084dd17c4bcca1d78080138b72da&pm=s) 

I’m quite curious why you’re seeing the need to create also a calendar component? Because I think the current problem is not, that we need just another calendar integration.

On top of that, I would like to share some thoughts with you – maybe they can be an inspiration to you.
We’re having a calendar in our App and I researched for that a lot of libraries. 

Atm we’re using https://demo.mobiscroll.com/react/eventcalendar/mobile-month-view# because it’s quite feature rich. The big downside is, that it’s a property driven library and not composable. That means it’s hard to customize it, and you need to learn their opinionated way of doing it (Which is still quite limited). This problem I saw also for a lot of other calendar libraries.

The problem is quite good explained in this [Video](https://youtu.be/Cpc9H4GLEno?t=669) based on a Graph library starting at Minute 11:09

Having now a composable approach would give me superpowers to customize the calendar the way I need it.
#### One example use case - Composable approach:
Let’s say we have the classic calendar day view
![image](https://user-images.githubusercontent.com/87228994/213665885-32482958-8d72-42b0-ad21-491932c08c8b.png)
And now I need to display only one event somewhere else
![image](https://user-images.githubusercontent.com/87228994/213665945-1a32c603-e3f5-4787-9359-9f28a8a92a19.png)

```jsx
<EventCard>
  <View direction="column" align="start">
    <Text variant="body-strong-1">{event.headline}</Text>
    <Text variant="body-2">{event.time}</Text>
  </View>
</EventCard>
```

Or alternatively, maybe I want to compose it like this:
![image](https://user-images.githubusercontent.com/87228994/213666125-89fd12cb-a257-4906-9d9d-67c854da9c0c.png)

```jsx
<View direction="row" gap={2}>
  {
    events.map(event => {
      return (
        <EventCard>
          <View direction="column" align="start">
            <Text variant="body-strong-1">{event.headline}</Text>
            <Text variant="body-2">{event.time}</Text>
          </View>
        </EventCard>
      )
    })
  }
</View>
```

With a composable approach, this would be quite easy to achieve. With the libraries on the market that I have seen, it’s not possible out of the box.

#### In contrast mobiscroll approach

```jsx
<Eventcalendar
            theme="ios" 
            themeVariant="light"
            clickToCreate={true}
            dragToCreate={true}
            dragToMove={true}
            dragToResize={true}
            eventDelete={true}
            locale={localeDe}
            data={myEvents}
            view={view}
            onEventClick={onEventClick}
       />
```

---

So I think having a calendar App that would be composable, could have a real USP.

As always, curious about your thoughts :)

**Comment by blvdmitry** [comment]
Hey, I think this is a good example. Reminds me of libraries like MUI selling advanced components so definitely a USP, similar to components like data grid. Having it available as a headless or a truly compound component can definitely help with the composition even for such advanced cases. 

I'm going to include this case into the calendar ticket as well to at least research what are the limitations of this approach. At the same time I would say the biggest priority right now from my side is to make sure the library is ready for quickly adopting features that are going to be released on the next Figma config and making all the planned changes around the available component variants, design tokens and their flexibility. 

Figma Config usually runs early May, so probably we'll be done with all the changes by June and that's when we can start looking into the new advanced components topics. Sorry if that sounds like a long timeline

**Reply by mmailaender** [reply]
The timeline is absolutely fine - I agree totally that there are currently more important issues - I was only seeing this Story on your notion Board, and wanted to drop some Inspiration 🙂
